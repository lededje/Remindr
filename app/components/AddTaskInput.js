import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  TextInput,
} from 'react-native';
import autobind from 'autobind-decorator';

export default class AddTaskInput extends Component {

  static propTypes = {
    onSubmit: React.PropTypes.func,
  }

  constructor(props) {
    super(props);

    this.state = {
      text: '',
    };
  }

  @autobind
  onSubmit() {
    this.props.onSubmit({ title: this.state.text });
    this.setState({ text: '' });
  }

  @autobind
  onChange(event) {
    this.setState({ text: event.nativeEvent.text });
  }

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          enablesReturnKeyAutomatically={true}
          blurOnSubmit={true}
          autoCapitalize="sentences"
          onSubmitEditing={this.onSubmit}
          onChange={this.onChange}
          value={this.state.text}
          testID="Add Task"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    borderTopWidth: 1,
    borderTopColor: '#d9dbdb',
    padding: 10,
  },
  input: {
    height: 19,
  },
});
