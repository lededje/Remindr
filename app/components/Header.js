import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Platform,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    borderColor: '#d9dbdb',
    borderBottomWidth: 1,
    paddingTop: Platform.OS === 'ios' ? 20 : 0,
  },
  title: {
    fontSize: 29,
    fontWeight: '300',
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 32,
    paddingRight: 32,
  },
});

export default function Header() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Remindr</Text>
    </View>
  );
}
