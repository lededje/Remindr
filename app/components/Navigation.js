import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
} from 'react-native';

import colors from '../util/colors';

export default function Navigation(props) {
  const types = ['DEFERRED', 'CURRENT', 'DONE'];

  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        {types.map((type) => {
          const selecteClassName = styles[type.toLowerCase()];
          const buttonStyles = [styles.button];

          if (type === props.selectedType) {
            buttonStyles.push(selecteClassName, styles.selected);
          }

          return (
            <TouchableHighlight
              // This is the most elegent way of binding the callback with the value.
              // eslint-disable-next-line react/jsx-no-bind
              onPress={props.onPress.bind(null, type)}
              key={type}
              accessibilityLabel={`navigation-${type.toLowerCase()}`}
            >
              <Text style={buttonStyles}>{type}</Text>
            </TouchableHighlight>
          );
        })}
      </View>
    </View>
  );
}

Navigation.propTypes = {
  onPress: React.PropTypes.func.isRequired,
  selectedType: React.PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    borderTopColor: '#d9dbdb',
    borderTopWidth: 1,
    backgroundColor: '#fff',
  },
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    padding: 12,
    fontSize: 11,
    fontWeight: '300',
  },

  // The selected styles

  deferred: {
    backgroundColor: colors.ORANGE,
  },
  current: {
    backgroundColor: colors.BLUE,
  },
  done: {
    backgroundColor: colors.GREEN,
  },
  selected: {
    color: '#fff',
  },
});
