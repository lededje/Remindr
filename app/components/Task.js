import React from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';

export default function Task(props) {
  return (
    <View style={styles.item} testID="Task">
      <Text style={styles.title}>{props.title}</Text>
      <Text style={styles.timestamp}>{props.timestamp}</Text>
    </View>
  );
}

Task.propTypes = {
  id: React.PropTypes.number.isRequired,
  title: React.PropTypes.string.isRequired,
  timestamp: React.PropTypes.number.isRequired,
};

const styles = StyleSheet.create({
  item: {
    paddingLeft: 35,
    paddingRight: 35,
    paddingTop: 20,
    paddingBottom: 20,
    backgroundColor: '#fff',
    borderColor: '#eee',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  title: {
    fontSize: 14,
    marginBottom: 10,
    fontWeight: '300',
  },
  timestamp: {
    fontSize: 12,
    color: '#999',
    fontWeight: '300',
  },
});
