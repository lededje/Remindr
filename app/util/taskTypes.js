import icons from './icons';
import colors from './colors';

const taskTypes = {
  CONFIGURE: {
    id: 'CONFIGURE',
    name: 'Configure',
    color: colors.ORANGE,
    icon: icons.SPANNER,
  },
  DEFERRED: {
    id: 'DEFERRED',
    name: 'Deferred',
    color: colors.ORANGE,
    icon: icons.CLOCK,
    sortBy: 'deferredUntil',
    order: 'asc',
  },
  CURRENT: {
    id: 'CURRENT',
    name: 'Current',
    color: colors.BLUE,
    icon: icons.CHECKBOX,
    sortBy: 'timestamp',
    order: 'asc',
  },
  DONE: {
    id: 'DONE',
    name: 'Complete',
    color: colors.GREEN,
    icon: icons.TICK,
    sortBy: 'timestamp',
    oder: 'desc',
  },
  DELETED: {
    id: 'DELETED',
    name: 'Deleted',
    color: colors.RED,
    icon: icons.CROSS,
  },
};

const flow = [
  taskTypes.CONFIGURE,
  taskTypes.DEFERRED,
  taskTypes.CURRENT,
  taskTypes.DONE,
  taskTypes.DELETED,
];

export { taskTypes as default, flow };
