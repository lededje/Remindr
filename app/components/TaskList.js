import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ListView,
  Text,
} from 'react-native';
import Task from './Task';

export default class List extends Component {

  static propTypes = {
    tasks: React.PropTypes.arrayOf(React.PropTypes.shape({
      id: React.PropTypes.number.isRequired,
      title: React.PropTypes.string.isRequired,
      timestamp: React.PropTypes.number.isRequired,
    })).isRequired,
  };

  static defaultProps = {
    tasks: [],
  }

  constructor(props) {
    super(props);

    this.dataSource = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
  }

  renderRow(task) {
    return (
      <Task id={task.id} title={task.title} timestamp={task.timestamp} />
    );
  }

  render() {
    const dataSource = this.dataSource.cloneWithRows(this.props.tasks);

    return (
      <View style={styles.container}>
        {this.props.tasks.length === 0 && (
          <ListView
            ref={(lv) => { this.listView = lv; }}
            dataSource={dataSource}
            renderRow={this.renderRow}
          />
        )}
        {this.props.tasks.length > 0 && (
          <Text>This list is empty</Text>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
