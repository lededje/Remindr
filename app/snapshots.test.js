import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import mockdate from 'mockdate';

import taskTypes from './util/taskTypes';

import AddTaskInput from './components/AddTaskInput';
import Navigation from './components/Navigation';
import Header from './components/Header';
import TaskList from './components/TaskList';
import Task from './components/Task';

const time = '2015-10-21T16:29:00Z';

describe('Component snapshots', () => {
  beforeEach(() => {
    mockdate.set(time);
  });

  afterEach(() => {
    mockdate.reset();
  });

  it('renders the component', () => {
    const component = renderer.create(
      <AddTaskInput />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the navigation component', () => {
    const dummyPressCallback = () => '';
    const component = renderer.create(
      <Navigation selectedType="CURRENT" onPress={dummyPressCallback} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the header component', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the task list component', () => {
    const taskData = [{
      id: 0,
      title: 'Test task 1',
      timestamp: moment(time).subtract(2, 'hours').format(),
      type: 'CURRENT',
    }, {
      id: 1,
      title: 'Test task 2',
      timestamp: moment(time).subtract(2, 'hours').format(),
      type: 'CURRENT',
    }];

    const component = renderer.create(
      <TaskList tasks={taskData} left={taskTypes.DEFERRED} right={taskTypes.DONE} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an empty list component with message', () => {
    const taskData = [];
    const component = renderer.create(
      <TaskList tasks={taskData} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Calendar time causes this to use Date.now to calculate result so breaks

  it('renders a current task component', () => {
    const component = renderer.create(
      <Task
        id={123}
        type="CURRENT"
        title="Current Task"
        timestamp={moment(time).subtract(2, 'hours').format()}
        left={taskTypes.DEFERRED}
        right={taskTypes.DONE}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('renders a deferred task component', () => {
    const component = renderer.create(
      <Task
        id={123}
        type="DEFERRED"
        title="Deferred Task"
        timestamp={moment(time).subtract(2, 'hours').format()}
        deferredUntil={moment(time).add(2, 'hours').format()}
        right={taskTypes.CURRENT}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a done task component', () => {
    const component = renderer.create(
      <Task
        id={123}
        type="DONE"
        title="Done Task"
        timestamp={moment(time).subtract(2, 'hours').format()}
        left={taskTypes.CURRENT}
        right={taskTypes.DELETE}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
