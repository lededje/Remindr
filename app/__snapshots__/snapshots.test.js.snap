exports[`Component snapshots renders a current task component 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#8bc34a",
      "height": undefined,
      "position": "relative",
    }
  }>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#eee",
        "height": undefined,
        "paddingBottom": 20,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
    testID="Task">
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "300",
          "marginBottom": 10,
        }
      }>
      Current Task
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999",
              "fontSize": 12,
              "fontWeight": "300",
            },
            Object {
              "flex": 1,
            },
          ]
        }>
        2 hours ago
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component snapshots renders a deferred task component 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#03a9f4",
      "height": undefined,
      "position": "relative",
    }
  }>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#eee",
        "height": undefined,
        "paddingBottom": 20,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
    testID="Task">
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "300",
          "marginBottom": 10,
        }
      }>
      Deferred Task
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999",
              "fontSize": 12,
              "fontWeight": "300",
            },
            Object {
              "flex": 1,
            },
          ]
        }>
        in 6 hours
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999",
                "fontSize": 12,
                "fontWeight": "300",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "GLYPHICONS Halflings",
              },
              Object {
                "marginRight": 4,
              },
            ]
          }>
          î„£
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999",
                "fontSize": 12,
                "fontWeight": "300",
              },
            ]
          }>
          Tomorrow at 5:29 PM
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component snapshots renders a done task component 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "height": undefined,
      "position": "relative",
    }
  }>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#eee",
        "height": undefined,
        "paddingBottom": 20,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
    testID="Task">
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "300",
          "marginBottom": 10,
        }
      }>
      Done Task
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999",
              "fontSize": 12,
              "fontWeight": "300",
            },
            Object {
              "flex": 1,
            },
          ]
        }>
        in 6 hours
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component snapshots renders an empty list component with message 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 20,
        "textAlign": "center",
      }
    }>
    This list is empty
  </Text>
</View>
`;

exports[`Component snapshots renders the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderTopColor": "#d9dbdb",
      "borderTopWidth": 1,
      "padding": 10,
    }
  }>
  <TextInput
    autoCapitalize="sentences"
    blurOnSubmit={true}
    enablesReturnKeyAutomatically={true}
    onChange={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "height": 19,
      }
    }
    testID="Add Task"
    value="" />
</View>
`;

exports[`Component snapshots renders the header component 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#d9dbdb",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 29,
        "fontWeight": "300",
        "paddingBottom": 20,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 20,
      }
    }>
    Remindr
  </Text>
</View>
`;

exports[`Component snapshots renders the navigation component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderTopColor": "#d9dbdb",
      "borderTopWidth": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="navigation-deferred"
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12,
            },
          ]
        }>
        DEFERRED
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="navigation-current"
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12,
            },
            Object {
              "backgroundColor": "#03a9f4",
            },
            Object {
              "color": "#fff",
            },
          ]
        }>
        CURRENT
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="navigation-done"
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12,
            },
          ]
        }>
        COMPLETE
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component snapshots renders the task list component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#8bc34a",
          "height": undefined,
          "position": "relative",
        }
      }>
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#eee",
            "height": undefined,
            "paddingBottom": 20,
            "paddingLeft": 35,
            "paddingRight": 35,
            "paddingTop": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
        testID="Task">
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "300",
              "marginBottom": 10,
            }
          }>
          Test task 1
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#999",
                  "fontSize": 12,
                  "fontWeight": "300",
                },
                Object {
                  "flex": 1,
                },
              ]
            }>
            in 6 hours
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#8bc34a",
          "height": undefined,
          "position": "relative",
        }
      }>
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#eee",
            "height": undefined,
            "paddingBottom": 20,
            "paddingLeft": 35,
            "paddingRight": 35,
            "paddingTop": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
        testID="Task">
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "300",
              "marginBottom": 10,
            }
          }>
          Test task 2
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#999",
                  "fontSize": 12,
                  "fontWeight": "300",
                },
                Object {
                  "flex": 1,
                },
              ]
            }>
            in 6 hours
          </Text>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
